test/widgets/content_test.dart

```diff
@@ -290,6 +297,79 @@ void main() {
 
   testContentSmoke(ContentExample.mathBlock);
 
+  testContentSmoke(ContentExample.strong);
+
+  testContentSmoke(ContentExample.emphasis);
+
+  group('InlineCodeNode', () {
+    testContentSmoke(ContentExample.inlineCode);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      /// Simulate a nested span's style by merging all ancestor-span styles,
+      /// starting from the root.
+      ///
+      /// This isn't how the style actually gets applied in the code under test,
+      /// but it should hopefully simulate it closely enough.
+      TextStyle? mergeStylesOuterToInner(InlineSpan outerSpan, InlineSpan innerSpan) {
+        final styles = <TextStyle?>[];
+        bool recurse(InlineSpan span) {
+          styles.add(span.style);
+          if (span == innerSpan) {
+            return false;
+          }
+          final notInSubtree = span.visitDirectChildren(recurse);
+          if (notInSubtree) {
+            styles.removeLast();
+          }
```

Comment:
Here's a variant that's slightly more efficient:
```suggestion
          if (span == innerSpan) {
            styles.add(span.style);
            return false;
          }
          final notInSubtree = span.visitDirectChildren(recurse);
          if (!notInSubtree) {
            styles.add(span.style);
          }
```

This way we only start adding to the list when we find the inner span, and never add anything we later have to remove.

The list goes from inner to outer when built this way, rather than outer to inner; we can then say `styles.reversed.reduce(…)` below.